### PR TITLE
Update pass on documentation and README.md

### PR DESCRIPTION
**Pull request by danthe3rd**
## What does this PR do?
Adds more documentation (especially about `xformers.ops`) and remove outdated documentation.

* Simplified installation explanation
* Removed instructions on how to install functorch ([now part of pytorch](https://pytorch.org/blog/PyTorch-1.13-release/))
* Removed doc on `factory` from README, as we are not supporting that on the newer operators, and it should be used mostly for internal testing (it is still part of the doc tho)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: xformers/ops/memory_efficient_attention.py [hunk 1]
```diff
@@ -22,17 +22,47 @@
 
 
 class AttentionMask:
```
Comment (fmassa) [inline xformers/ops/memory_efficient_attention.py:24]: Nit: I though this attention mask class was temporary until we merged it with the other AttentionMask available? It would be good to make this distinction clear at some point, otherwise it will be very confusing

**Review by fmassa — APPROVED** [review]
Looks great, thanks!

File: README.md [hunk 2]
```diff
@@ -19,115 +19,52 @@
...
-xFormers is a modular and field agnostic library to flexibly generate transformer architectures from interoperable and optimized building blocks. These blocks are not limited to xFormers and can also be cherry picked as the user see fit.
-
-## Getting started
-
-The full [documentation](https://facebookresearch.github.io/xformers/) contains instructions for getting started, deep dives and tutorials about the various APIs.
-If in doubt, please check out the [HOWTO](HOWTO.md). Only some general considerations are laid out in the README.
+## xFormers - Toolbox to Accelerate Research on Transformers
 
-For recent changes, you can have a look at the [changelog](CHANGELOG.md)
+xFormers is:
```
Comment (blefaudeux) [inline README.md:24]: looks nice, great wording IMO

File: README.md [hunk 3]
```diff
@@ -19,115 +19,52 @@
...
-
-<details><summary> Triton </summary><p>
-
-Some parts of xFormers use [Triton](http://www.triton-lang.org), and will only expose themselves if Triton is installed, and a compatible GPU is present (nVidia GPU with tensor cores). If Triton was not installed as part of the testing procedure, you can install it directly by running `pip install triton`. You can optionally test that the installation is successful by running one of the Triton-related benchmarks, for instance `python3 xformers/benchmarks/benchmark_triton_softmax.py`
-
-Triton will cache the compiled kernels to `/tmp/triton` by default. If this becomes an issue, this path can be specified through the `TRITON_CACHE_DIR` environment variable.
-
-</p></details>
+```bash
+# (Optional) Makes the build much faster
```
Comment (blefaudeux) [inline README.md:40]: maybe a word on memory consumption for some machines ? I kind of remember people getting OOMs there (though I would definitely recommend ninja of course)

File: README.md [hunk 4]
```diff
@@ -19,115 +19,52 @@
...
+pip install ninja
+# Set TORCH_CUDA_ARCH_LIST if running and building on different GPU types
```
Comment (blefaudeux) [inline README.md:42]: useful addendum..

File: README.md [hunk 5]
```diff
@@ -288,11 +229,11 @@ xFormers has a BSD-style license, as found in the [LICENSE](LICENSE) file.
 If you use xFormers in your publication, please cite it by using the following BibTeX entry.
 
 ``` bibtex
-@Misc{xFormers2021,
-  author =       {Benjamin Lefaudeux and Francisco Massa and Diana Liskovich and Wenhan Xiong and Vittorio Caggiano and Sean Naren and Min Xu and Jieru Hu and Marta Tintore and Susan Zhang},
+@Misc{xFormers2022,
+  author =       {Benjamin Lefaudeux and Francisco Massa and Diana Liskovich and Wenhan Xiong and Vittorio Caggiano and Sean Naren and Min Xu and Jieru Hu and Marta Tintore and Susan Zhang and Patrick Labatut and Daniel Haziza},
```
Comment (blefaudeux) [inline README.md:233]: feel free to change the author ordering anytime here, I won't be offended personally. Any metrics or semi-reasonable logic works, could be by alphabetical order or contributions for instance, anything which works for you

File: xformers/ops/memory_efficient_attention.py [hunk 6]
```diff
@@ -733,16 +822,71 @@ def memory_efficient_attention(
     attn_bias: Optional[Union[torch.Tensor, AttentionMask]] = None,
     p: float = 0.0,
     *,
-    op=None,
-):
-    """
-    Implements the memory-efficient attention mechanism following
+    op: Optional[AttentionOp] = None,
+) -> torch.Tensor:
+    """Implements the memory-efficient attention mechanism following
```
Comment (blefaudeux) [inline xformers/ops/memory_efficient_attention.py:827]: super comprehensive and on point, thanks @danthe3rd. I tend to favor doc in the code, doesn't rot as fast, and this looks as good as it gets

**Review by blefaudeux — APPROVED** [review]
looks great to  me, definite improvement, thanks @danthe3rd

File: README.md (hunk 5 above)
Comment (danthe3rd) [reply README.md:233]: We can keep it like this for now, and update ~every year when we have new contributors

File: README.md (hunk 3 above)
Comment (danthe3rd) [reply README.md:40]: Added that to the troubleshooting section - thanks !

File: xformers/ops/memory_efficient_attention.py (hunk 1 above)
Comment (danthe3rd) [reply xformers/ops/memory_efficient_attention.py:24]: Let's do that as part of another PR as I'm not very familiar with the other attention masks yet
